woocommerce/woocommerce
# CYS: Insert Add Patterns button for no blocks placeholder

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This feature inserts an "Add pattern" button to the No Blocks view which is rendered when a user removes all the patterns in the assembler.

There are a few notable changes that come with this:
* I've extracted the pattern insertion logic into its own hook (`useInsertPattern`) so I can reuse it for this feature.
* I've created a new hook called `useInsertPatternByName` enabling you to pass the function a pattern slug and it will insert it into the editor.

Open questions:
* I've inserted the default pattern by name when the user clicks the button, is this an OK approach or is there another approach that would be better and not rely on us knowing the pattern name ahead of time? There is a level of safety here since if anything changes regarding the pattern name which breaks the experience it will get caught by E2E tests.





Closes https://github.com/woocommerce/woocommerce/issues/49033



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Start over with the customize your store experience
2. Click into "Design your homepage"
3. Remove all patterns and ensure the "Add pattern" button appears, check that clicking it should insert a default pattern.
4. Do some regression testing around inserting/removing patterns to check no regressions were introduced when extracting the insert pattern logic into its own hook.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
